### ISSUE TITLE
Add support for triggering ad-hoc snapshots in Debezium

### ISSUE THREAD

**Issue by gunnarmorling** [kcctl/kcctl#390]
Debezium provides the ability to run ad-hoc (incremental) snapshots while a connector is running, allowing for instance to backfill the data for a specific table whose corresponding Kafka topic got accidentially deleted. Snapshots are triggered via [signals](https://debezium.io/documentation/reference/stable/configuration/signalling.html#sending-signals-to-a-debezium-connector). It would be interesting to see what it'd take to do so using kcctl.

This would broaden the scope of kcctl quite a bit, as that's a connector specific command, rather than generic Kafka Connect level functionality. I believe this could be very useful though, when done the right way (ideally, in some sort of pluggable/extensible way).

**Comment by gunnarmorling** [comment]
@C0urante, very curious about your take here :)

**Comment by mfvitale** [comment]
Hi @gunnarmorling, this is just my thought, but what do you think if the core kcctl functionalities can be provided as a library so that it can be reused by connector specific `ctl`?

This can open the possibility to create for example a `debctl` with less effort reusing the common Connect API and just implement the specific commands.

**Comment by gunnarmorling** [comment]
Hey @mfvitale, sorry for the late reply. I'm not quite sure right now how much library functions we actually could extract here? I think picocli essentially is the common part one could also use to build `dbzctl` if one wanted to? That being said, do you happen to know whether and if so how `kubectl` is extensible? Maybe something to draw inspiration from?

**Comment by mfvitale** [comment]
> I think picocli essentially is the common part one could also use to build dbzctl if one wanted to?

Yes, I am referring to all Connect commands that should also have a sense for a hypothetic `dbzctl`.  Just to avoid code duplication and reimplement the same. 

> That being said, do you happen to know whether and if so how kubectl is extensible? Maybe something to draw inspiration from?

This is another way to see it: extensibility. Maybe a good start is https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/
